openmdao/core/driver.py

```diff
@@ -106,19 +108,115 @@ def _setup_driver(self, problem):
         self._problem = problem
         model = problem.model
 
+        self._objs = objs = OrderedDict()
+        self._cons = cons = OrderedDict()
+        self._responses = model.get_responses(recurse=True)
+        for name, data in iteritems(self._responses):
+            if data['type'] == 'con':
+                cons[name] = data
+            else:
+                objs[name] = data
+
         # Gather up the information for design vars.
         self._designvars = model.get_design_vars(recurse=True)
 
-        self._responses = model.get_responses(recurse=True)
-        self._objs = model.get_objectives(recurse=True)
-        self._cons = model.get_constraints(recurse=True)
+        con_set = set()
+        obj_set = set()
+        dv_set = set()
+
+        self._remote_dvs = dv_dict = {}
+        self._remote_cons = con_dict = {}
```

Comment:
Looks like a few vars to add to the attributes in the class docstring.

Reply:
fixed